### PR TITLE
Allow shadows to be colored when physical layer is drawn by engine

### PR DESCRIPTION
**Pull request by jonahwilliams**
When investigating https://github.com/flutter/flutter/issues/15641 I found two issues with shadows drawn by a physical layer. 

  * shadows do not match between engine and flutter.
  * shadow color is ignored by engine

The physical layer created by a `Material` widget can either be painted in the flutter package or by the engine, depending on whether the `Material` widget needs compositing.

In [`RenderPhysicalShape.paint`](https://github.com/flutter/flutter/blob/9c49255f3eeb4873e6398c805b4bb6b9737be962/packages/flutter/lib/src/rendering/proxy_box.dart#L1594), a colored shadow is painted if `needsCompositing` is false.  Below is a widget with an elevated Material and a slightly red shadow color.

source code: https://gist.github.com/jonahwilliams/26b63c58efd8e899bd3495047864cea3

 

Same setup, but with default colors.  The shadow appears to change slightly - I'm not sure which one is correct but they should be synchronized. 

## With fix
The shadow is the same color whether or not the engine or flutter drew the layer.  shadows are still slightly different.

### CODE REVIEW

File: shell/platform/android/io/flutter/view/AccessibilityBridge.java [hunk 1]
```diff
@@ -138,7 +138,7 @@ public AccessibilityNodeInfo createAccessibilityNodeInfo(int virtualViewId) {
             result.setAccessibilityFocused(mA11yFocusedObject.id == virtualViewId);
 
         if (object.hasFlag(Flag.IS_TEXT_FIELD)) {
-            result.setPassword(object.hasFlag(Flag.IS_PASSWORD);
+            result.setPassword(object.hasFlag(Flag.IS_PASSWORD));
```
Comment (jason-simmons) [inline shell/platform/android/io/flutter/view/AccessibilityBridge.java]: this is already fixed in the tip of tree

Comment (jonahwilliams) [reply shell/platform/android/io/flutter/view/AccessibilityBridge.java]: Fixed

File: lib/ui/compositing.dart [hunk 2]
```diff
@@ -133,10 +133,10 @@ class SceneBuilder extends NativeFieldWrapperClass2 {
   /// Rasterization will be clipped to the given shape.
   ///
   /// See [pop] for details about the operation stack.
```
Comment (Hixie) [inline lib/ui/compositing.dart]: Please document the arguments in the docs here

File: lib/ui/compositing.dart [hunk 3]
```diff
@@ -133,10 +133,10 @@ class SceneBuilder extends NativeFieldWrapperClass2 {
...
-  void pushPhysicalShape({ Path path, double elevation, Color color }) {
-    _pushPhysicalShape(path, elevation, color.value);
+  void pushPhysicalShape({ Path path, double elevation, Color color, Color shadowColor}) {
+    _pushPhysicalShape(path, elevation, color.value, shadowColor?.value ?? 0);
```
Comment (Hixie) [inline lib/ui/compositing.dart]: is 0 the right default?

**Comment by Hixie** [comment]
LGTM

cc @chinmaygarde for engine review

File: lib/ui/compositing.dart (hunk 3 above)
Comment (jonahwilliams) [reply lib/ui/compositing.dart]: Ahh no, should be SK_ColorBLACK https://github.com/google/skia/blob/master/include/core/SkColor.h#L68 - Fixed

File: lib/ui/compositing.dart (hunk 2 above)
Comment (jonahwilliams) [reply lib/ui/compositing.dart]: Done

**Pull request merged**